src/Testcontainers.InfluxDb/InfluxDbBuilder.cs

```diff
@@ -0,0 +1,168 @@
+    {
+        return Merge(DockerResourceConfiguration, new InfluxDbConfiguration(password: password))
+            .WithEnvironment("DOCKER_INFLUXDB_INIT_PASSWORD", password);
+    }
+
+    /// <summary>
+    /// Sets the InfluxDb organization.
+    /// </summary>
+    /// <param name="organization">The InfluxDb organization.</param>
+    /// <returns>A configured instance of <see cref="InfluxDbBuilder" />.</returns>
+    public InfluxDbBuilder WithOrganization(string organization)
+    {
+        return Merge(DockerResourceConfiguration, new InfluxDbConfiguration(organization: organization))
+            .WithEnvironment("DOCKER_INFLUXDB_INIT_ORG", organization);
+    }
+
+    /// <summary>
+    /// Sets the InfluxDb bucket.
+    /// </summary>
+    /// <param name="bucket">The InfluxDb bucket.</param>
+    /// <returns>A configured instance of <see cref="InfluxDbBuilder" />.</returns>
+    public InfluxDbBuilder WithBucket(string bucket)
+    {
+        return Merge(DockerResourceConfiguration, new InfluxDbConfiguration(bucket: bucket))
+            .WithEnvironment("DOCKER_INFLUXDB_INIT_BUCKET", bucket);
+    }
+
+    /// <summary>
+    /// Sets the InfluxDb admin token.
+    /// </summary>
+    /// <param name="adminToken">The InfluxDb admin token.</param>
+    /// <returns>A configured instance of <see cref="InfluxDbBuilder" />.</returns>
+    public InfluxDbBuilder WithAdminToken(string adminToken)
+    {
+        return Merge(DockerResourceConfiguration, new InfluxDbConfiguration(adminToken: adminToken))
+            .WithEnvironment("DOCKER_INFLUXDB_INIT_ADMIN_TOKEN", adminToken);
+    }
+
+    /// <summary>
+    /// Sets the InfluxDb retention.
+    /// </summary>
+    /// <param name="retention">The InfluxDb retention.</param>
+    /// <returns>A configured instance of <see cref="InfluxDbBuilder" />.</returns>
+    public InfluxDbBuilder WithRetention(string retention)
+    {
+        return Merge(DockerResourceConfiguration, new InfluxDbConfiguration(retention: retention))
+            .WithEnvironment("DOCKER_INFLUXDB_INIT_RETENTION", retention);
+    }
+
+    /// <inheritdoc />
+    public override InfluxDbContainer Build()
+    {
+        Validate();
+        return new InfluxDbContainer(DockerResourceConfiguration, TestcontainersSettings.Logger);
+    }
+
+    /// <inheritdoc />
+    protected override InfluxDbBuilder Init()
+    {
+        return base.Init()
+            .WithImage(InfluxDbImage)
+            .WithPortBinding(InfluxDbPort, true)
+            .WithEnvironment("DOCKER_INFLUXDB_INIT_MODE", "setup")
+            .WithUsername(DefaultUsername)
+            .WithPassword(DefaultPassword)
+            .WithOrganization(DefaultOrganization)
+            .WithBucket(DefaultBucket)
+            .WithWaitStrategy(Wait.ForUnixContainer().UntilHttpRequestIsSucceeded(request =>
+                request.ForPort(InfluxDbPort).ForPath("/ping").ForStatusCode(HttpStatusCode.NoContent)));
```

Comment:
Doesn't Java set up basic HTTP authentication using the username and password configuration? Is it not necessary?

Reply:
For `/ping` endpoint as far as I understand it is not required to provide any token.
https://docs.influxdata.com/influxdb/v2.7/api/#operation/GetPing

Reply:
> The influx ping command checks the health of a running InfluxDB instance by querying the /health endpoint. It does not require an API token.

You are right, thanks for double-checking.